### PR TITLE
CDS-74797 Runtime Issue Key

### PR DESCRIPTION
**Pull request by michaelcretzman**
https://harness.atlassian.net/browse/CDS-74797

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ x] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* https://harness.atlassian.net/browse/CDS-74797

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages.md [hunk 1]
```diff
@@ -37,7 +37,10 @@ You can also [update Jira issues](/docs/continuous-delivery/x-platform-cd-featur
 
 In **Issue Key**, you can use an expression to reference the Key ID from another Jira Create or Jira Update step.
 
-The Jira Create or Jira Update step you want to reference must be before the Jira Update step that references it in the stage.
+Here are some important consideration when using an expression:
```
Comment (michaelcretzman) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages.md:40]:
```suggestion
Here are some important considerations when using an expression in **Issue Key:**
```

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages.md [hunk 2]
```diff
@@ -37,7 +37,10 @@ You can also [update Jira issues](/docs/continuous-delivery/x-platform-cd-featur
...
+
+- The Jira Create or Jira Update step you want to reference must be before the Jira Update step that references it in the stage.
+- Harness uses the **Jira Project** and **Issue Type** you in **Add Jira Fields** to fetch the list of fields. If you use a runtime expression for **Issye Key** and the Jira project and issue type you selected does not produce the issue key provided at runtime, your pipeline could fail to execute successfully.
```
Comment (michaelcretzman) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages.md:43]:
```suggestion
- Harness uses the **Jira Project** and **Issue Type** you select in **Add Jira Fields** to fetch the list of fields. If you use a runtime expression for **Issue Key** and the Jira project and issue type you selected does not produce the issue key provided at runtime, your pipeline could fail to execute successfully. 
```

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages.md [hunk 3]
```diff
@@ -37,7 +37,10 @@ You can also [update Jira issues](/docs/continuous-delivery/x-platform-cd-featur
 
 In **Issue Key**, you can use an expression to reference the Key ID from another Jira Create or Jira Update step.
```
Comment (naman-goenka) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages.md:38]: This feature is also applicable when issue key is `runtime input`

File: docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages.md [hunk 4]
```diff
@@ -37,7 +37,10 @@ You can also [update Jira issues](/docs/continuous-delivery/x-platform-cd-featur
...
 
-The Jira Create or Jira Update step you want to reference must be before the Jira Update step that references it in the stage.
+Here are some important considerations when using an expression in **Issue Key:**
+
+- The Jira Create or Jira Update step you want to reference must be before the Jira Update step that references it in the stage.
+- Harness uses the **Jira Project** and **Issue Type** you select in **Add Jira Fields** to fetch the list of fields. If you use a runtime expression for **Issue Key** and the Jira project and issue type you selected does not produce the issue key provided at runtime, your pipeline could fail to execute successfully.
```
Comment (naman-goenka) [inline docs/continuous-delivery/x-platform-cd-features/cd-steps/ticketing-systems/update-jira-issues-in-cd-stages.md:43]: you selected does not `correspond with` the issue key provided

**Pull request merged**